src/_pytest/unittest.py

```diff
@@ -150,7 +149,7 @@ def cleanup(*args):
     def fixture(self, request: FixtureRequest) -> Generator[None, None, None]:
         if _is_skipped(self):
             reason = self.__unittest_skip_why__
-            pytest.skip(reason)
+            raise pytest.skip.Exception(reason, _use_item_location=True)
```

Comment:
i wonder if as a followup we should introduce a `for_item` parameter thats valid on all outcome exceptions to support the use-cases of skip/fail for a reason against  the/a item (and being able to ppass in a class/module instead of the direct item i necessary)

Reply:
The other two `OutcomeException`s are `Exit` and `Failed` (and `XFailed` subclass of `Failed`). For `Exit` I don't think it makes sense, and for `fail` I'm not sure either.